elementor/elementor
# Fix: Global colors not returning in `get_frontend_settings()`

## PR Checklist



## PR Type
What kind of change does this PR introduce?


## Summary

This PR can be summarized in the following changelog entry:

* fix: get_frontend_settings() not returning global color value

## Description
An explanation of what is done in this PR

In `includes/base/controls-stack.php` I've modified `get_frontend_settings()` method to check `__globals__` for the `frontend_available` controls and if they exists replaced the value with correct value. For now only check added is for color values. 

## Test instructions
This PR can be tested by following these steps:

* Add new color in your widget and make it available in frontend `'frontend_available' => true`
* From editor: set the color to any custom color.
* On front end you should get `data-settings="{"your_key":"hex_color_code"}" `
* Result: Correct.

* Again, From Editor: set the color to any global color.
* Before patch: you will get `data-settings="{"your_key":"default_color_you_defined"}" `
* After pacth: you should get `data-settings="{"your_key": "var(--e-global-color-YOUR_GLOBAL_COLOR)" }" `


## Quality assurance


Fixes #18291
